src/common/src/util/value_encoding/mod.rs

```diff
@@ -183,6 +183,14 @@ pub fn serialize_datum_into(datum_ref: impl ToDatumRef, buf: &mut impl BufMut) {
     }
 }
 
+pub fn estimate_serialize_datum_size(datum_ref: impl ToDatumRef) -> usize {
+    if let Some(d) = datum_ref.to_datum_ref() {
+        1 + estimate_serialize_scalar_size(d)
+    } else {
+        1
```

Comment:
I guess `1` here is to store the null bit flag?

Reply:
Yes